.github/workflows/cypress-workflow.yml

```diff
@@ -0,0 +1,92 @@
+      CI: 1
+      # avoid warnings like "tput: No value for $TERM and no -T specified"
+      TERM: xterm
+    steps:
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          # TODO: Parse this from alerting plugin
+          java-version: 14
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          path: alerting
+          # TODO: Move this after Kibana plugin setup so we can pull down the correct branch to support all opendistro branches
+          repository: opendistro-for-elasticsearch/alerting
+      - name: Run Elasticsearch with plugin
+        run: |
+          cd alerting
+          ./gradlew run &
+          timeout 300 bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' localhost:9200)" != "200" ]]; do sleep 5; done'
+      - name: Checkout Alerting Kibana plugin
+        uses: actions/checkout@v2
+        with:
+          path: alerting-kibana-plugin
+      - name: Get Kibana version
+        id: kibana_version
+        run: |
+          echo "::set-output name=kibana_version::$(node -p "(require('./alerting-kibana-plugin/kibana.json').kibanaVersion).match(/[.0-9]+/)[0]")"
+      - name: Checkout Kibana
+        uses: actions/checkout@v2
+        with:
+          repository: opendistro-for-elasticsearch/kibana-oss
+          ref: ${{ steps.kibana_version.outputs.kibana_version }}
+          token: ${{ secrets.GITHUB_KIBANA_OSS }}
+          path: kibana
+      - name: Get node and yarn versions
+        id: versions
+        run: |
+          echo "::set-output name=node_version::$(node -p "(require('./kibana/package.json').engines.node).match(/[.0-9]+/)[0]")"
+          echo "::set-output name=yarn_version::$(node -p "(require('./kibana/package.json').engines.yarn).match(/[.0-9]+/)[0]")"
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ steps.versions.outputs.node_version }}
+          registry-url: 'https://registry.npmjs.org'
+      - name: Install correct yarn version for Kibana
+        run: |
+          npm uninstall -g yarn
+          echo "Installing yarn ${{ steps.versions_step.outputs.yarn_version }}"
+          npm i -g yarn@${{ steps.versions.outputs.yarn_version }}
+      - name: Bootstrap Kibana with plugin
+        run: |
+          mkdir -p kibana/plugins
+          mv alerting-kibana-plugin kibana/plugins
+          cd kibana/plugins/alerting-kibana-plugin
+          yarn kbn bootstrap
+      - name: Run Kibana server
+        run: |
+          cd kibana
+          yarn start --no-base-path --no-watch &
+          sleep 300
+        # timeout 300 bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' localhost:5601/api/status)" != "200" ]]; do sleep 5; done'
+      - name: Run Cypress tests
+        uses: cypress-io/github-action@v2
+        with:
+          working-directory: kibana/plugins/alerting-kibana-plugin
+          command: yarn run cypress run
```

Comment:
Let's add `wait-on: 'http://localhost:5601'` after the command entry to ensure the endpoint is available before the tests.

Also, if `wait-on` is added does the `sleep 300` need to be included after `yarn start` above or was that added for another reason?

Reply:
There is a doubt if `wait-on` works in this situation, according to some past workflow runs.

Workflow runs with the first test case passed:
https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/467761518/workflow
`sleep 300` and `wait-on` both added (the `wait-on` timeout is 60 seconds by default).

https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/470299602/workflow
`Sleep 300` and without `wait-on`


Workflow runs with the first test case failed:
https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/469631264/workflow
`Sleep 240` and `wait-on` both added.

https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/469535842/workflow
`Sleep 180` and `wait-on` both added.

https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/467803060/workflow
`Sleep 120` and `wait-on` both added.

https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/467679972/workflow
`Wait-on` with a time-out for 360 seconds.

https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/467715838/workflow
`Wait-on` with a time-out for 480 seconds.

https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/actions/runs/470232057/workflow
Waiting using command "npx wait-on --timeout 30000 http-get://localhost:5601/app/opendistro-alerting"

Reply:
As a reference,
here is the "yarn command" for AD Kibana plugin to run the cypress test:
```
"test:e2e": "CHOKIDAR_USEPOLLING=1 CYPRESS_responseTimeout=180000 WAIT_ON_TIMEOUT=900000 start-server-and-test 'cd ../../ && yarn start --no-base-path --oss && cd plugins/anomaly-detection-kibana-plugin' http-get://localhost:5601/app/opendistro-anomaly-detection-kibana 'echo sleeping to wait for server to get ready && sleep 180 && yarn cy:run'"
```
(https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/master/package.json)

Reply:
The `sleep 300` was added to ISM Kibana plugin when upgrading to ES 7.9.1
https://github.com/opendistro-for-elasticsearch/index-management-kibana-plugin/commit/c4357005ebcb79b66bcde4540f725d2276104505

Reply:
I see, we can go ahead with these changes then, thanks for the links.

Reply:
No problem. I will update if I find a better way after experiment in the future. 😁